p2p/peer.go

```diff
@@ -511,9 +511,10 @@ type PeerInfo struct {
 // Info gathers and returns a collection of metadata known about a peer.
 func (p *Peer) Info() *PeerInfo {
 	// Gather the protocol capabilities
-	var caps []string
-	for _, cap := range p.Caps() {
-		caps = append(caps, cap.String())
+	peerCaps := p.Caps()
+	caps := make([]string, len(peerCaps))
+	for i, cap := range peerCaps {
+		caps[i] = cap.String()
```

Comment:
Please revert this. We almost never call this method, so it's runtime is irrelevant. The original code was a lot nicer to read/understand.

Reply:
Done